src/v/kafka/server/handlers/fetch.cc

```diff
@@ -421,6 +430,11 @@ static void fill_fetch_responses(
         range = boost::irange<size_t>(
           0, std::min({results.size(), responses.size()}));
     }
+
+    // Used to aggregate semaphore_units from results.
+    bool memory_units_unset = true;
+    read_result::memory_units_t total_memory_units;
```

Comment:
aka `std::optional`?

Reply:
Nice catch, need to stop coding after midnight lol. Switched it over to an optional.